stdlib/public/runtime/SwiftRT-ELF.cpp

```diff
@@ -63,8 +66,50 @@ static void swift_image_constructor() {
       SWIFT_SECTION_RANGE(swift4_assocty),
   };
 
-#undef SWIFT_SECTION_RANGE
-
   swift_addNewDSOImage(&sections);
 }
 
+#else // ELF_STATIC_LIB
+
+namespace swift {
+
+  void initializeProtocolLookup() {
+    const MetadataSections::Range protocols_range =
+      SWIFT_SECTION_RANGE(swift4_protocols);
+    if (protocols_range.length) {
+      const void *start = reinterpret_cast<void *>(protocols_range.start);
+      addImageProtocolsBlockCallback(start, protocols_range.length);
+    }
+  }
+
+  void initializeProtocolConformanceLookup() {
+    const MetadataSections::Range conformances_range =
+      SWIFT_SECTION_RANGE(swift4_protocol_conformances);
+    if (conformances_range.length) {
+      const void *start = reinterpret_cast<void *>(conformances_range.start);
+      addImageProtocolConformanceBlockCallback(start, conformances_range.length);
+    }
+  }
+
+  void initializeTypeMetadataRecordLookup() {
+    const MetadataSections::Range type_metadata_range =
+      SWIFT_SECTION_RANGE(swift4_type_metadata);
+    if (type_metadata_range.length) {
+      const void *start = reinterpret_cast<void *>(type_metadata_range.start);
+      addImageTypeMetadataRecordBlockCallback(start, type_metadata_range.length);
+    }
+  }
+
+  void initializeTypeFieldLookup() {
+    const MetadataSections::Range fields_range =
+      SWIFT_SECTION_RANGE(swift4_fieldmd);
+    if (fields_range.length) {
+      const void *start = reinterpret_cast<void *>(fields_range.start);
+      addImageTypeFieldDescriptorBlockCallback(start, fields_range.length);
+    }
+  }
+}
+
```

Comment:
Isn't this just a duplication of the `swift_addNewDSOImage`?  Why not just use that?